src/leveldbwrapper.cpp

```diff
@@ -67,6 +69,23 @@ CLevelDBWrapper::CLevelDBWrapper(const boost::filesystem::path& path, size_t nCa
     leveldb::Status status = leveldb::DB::Open(options, path.string(), &pdb);
     HandleError(status);
     LogPrintf("Opened LevelDB successfully\n");
+
+    bool got_key = Read(OBFUSCATE_KEY_KEY, obfuscate_key);
+
+    if (!got_key) {
+        if (obfuscate && IsEmpty()) {
+            // Initialize non-degenerate obfuscation if it won't upset 
+            // existing, non-obfuscated data.
+            obfuscate_key = CreateObfuscateKey();
+        } else {
+            // Our base-case for an obfuscation key is a zero'd byte array -- 
+            // this won't disrupt existing data.
+            obfuscate_key = std::string(OBFUSCATE_KEY_NUM_BYTES, '\000');
+        }
+
+        Write(OBFUSCATE_KEY_KEY, obfuscate_key);
+        LogPrintf("Wrote new obfuscate key '%s'\n", obfuscate_key);
```

Comment:
Hmm.. `obfuscate_key` is not necessarily printable. Maybe it would be better to log a hexified version of the key?